neuland/calibration/R3BNeulandCosmicTracker.cxx

```diff
@@ -67,12 +67,13 @@ namespace Neuland
         }
 
         CosmicTracker::CosmicTracker()
-            : fFit("CosmicTracker:fFit", "pol1")
+            : fDistances{ 0. }
+            , fFit("CosmicTracker:fFit", "pol1")
         {
             fDistances.reserve(MaxNumberOfPlanes);
-            for (auto p = 0; p < MaxNumberOfPlanes; ++p)
+            while (fDistances.size() < MaxNumberOfPlanes)
             {
-                fDistances[p] = p * BarSize_Z;
+                fDistances.push_back(fDistances.back() + BarSize_Z);
```

Comment:
I assume that CosmicTracker is instantiated only once per run. 

In that case, I prefer my variant (fDistances.resize(MaxNumberOfPlanes)) over your variant. 

I am aware that you are trying to avoid zero-initializing the vector only to overwrite the values again, but I find it less readable. (A larger fix would be to be to get rid of fDistances altogether.)

Reply:
Do you mean the following code?
```cpp
CosmicTracker::CosmicTracker()
            : fFit("CosmicTracker:fFit", "pol1")
        {
            fDistances.reserve(MaxNumberOfPlanes);
            fDistances.resize(MaxNumberOfPlanes);
            for (auto p = 0; p < MaxNumberOfPlanes; ++p)
            {
                fDistances[p] = p * BarSize_Z;
```

This has pointer arithmetic, which is not very preferable according to the core guideline:
> Bounds.1: Don’t use pointer arithmetic

Reply:
It seems your are correct in that the core guidelines also consider operator[] of (some) STL containers to be "pointer arithmetic". 

IMHO there is a world of a difference between std::vector::operator[] and adding offsets to raw pointers in that for the former I can simply enable boundary checks (which made this bug visible) and for the latter I can not. (Of course, this would then also affect line 470.)